test/client/fixtures/sandbox/code-instrumentation/process-script-test.js

```diff
@@ -78,3 +80,29 @@ test('should not process non-string args', function () {
     deepEqual(execScript('eval.apply(this, [[1, 2 ,3]])'), [1, 2, 3]);
 });
 
+module('eval assignment');
+
+test('the eval method should switch its own context to global', function () {
+    throws(function () {
+        execScript('(function(){ var t1 = "localScope", ev = eval; return ev("t1"); })()');
```

Comment:
Just thought what it would be better to use the following tactic in these tests:
Instead of expecting the throw, which may occur because of many different reasons, let's define var with the same name in the global scope, but with different value. So we will have the following test:
```js
 var actual = execScript(' window.t1 = 'globalScope'; var res = (function(){ var t1 = "localScope", ev = eval; return ev("t1"); })(); delete window.t1; res');

strictEqual(actual, 'globalScope');
```